redfish_client/resource.py

```diff
@@ -109,6 +117,45 @@ def execute_action(self, action_name, payload):
             return self._connector.post(action.target, payload=payload)
         raise KeyError("Action with {} does not exist".format(action_name))
 
+    def wait_for_stat(self, stat, expected_value, fail_values=list(), poll_every_num_sec=3,
+                      timeout=30*60):
+        """
+        :param stat: list of keys or single key
+        :param expected_value: expected value
+        :param fail_values: fail values
+        :param poll_every_num_sec: number of requests per second
+        :param timeout: timeout in seconds
+        :return: actual value if expected_value matches else return None
+        Raises:
+            TimedOutError: If timeout is exceeded.
+        """
+
+        if "@odata.id" not in self._content:
+            raise Exception("Element does not have '@odata.id' attribute, cannot wait for a stat "
+                            "inside inner object")
+
+        start_time = time.time()
+        time_delta = 0
+
+        if not isinstance(stat, list):
```

Comment:
Python 2 (if that's what we also want to support) doesn't allow variable args in function and keyword arguments with default value. 
We could check if instance is `collections.Iterable`, but that would fail for strings. We could also check for both `list` and `tuple` but then we miss out on other iterables.

We could make use of the `**kwargs` and get default values at the start of a function using `kwargs.get('poll_interval', 3)`

method definition would look something like this:

```
def wait_for_stat(self, expected, *stat, **kwargs):
    fails = kwargs.get('fails', ())
    poll_interval = kwargs.get('poll_interval', 3)
    timeout = kwargs.get('poll_interval', 30 * 60)
    ...
```
I just don't like that the first parameter has to be `expected`

Reply:
Or we can just avoid all this if we tell user that it has to be string; dot-syntax supported and we split it ourselves:

```
'Status.Substatus.Name' => ['Status', 'Substatus', 'Name']
```

Reply:
Since python2 is as good as dead at this stage, keeping our client backwards compatible is not high on my list of priorities. I would say keeping it working on python3 that is supplied by CentOS and Ubuntu LTS releases is good enough.

As for the parameter order, we can always rename the function to better fit the order of parameters. For example, `server.wait_for_value_at("my_value", "Status", "Something")` seems to be reasonable function name (not the best by far, but decent ;).